src/actix/api/snapshot_api.rs

```diff
@@ -72,6 +80,37 @@ async fn create_snapshot(
     process_response(response, timing)
 }
 
+#[post("/collections/{name}/snapshots/upload")]
+async fn upload_snapshot(
+    dispatcher: web::Data<Dispatcher>,
+    path: web::Path<String>,
+    MultipartForm(form): MultipartForm<SnapshottingForm>,
+    params: Query<SnapshottingParam>,
+) -> impl Responder {
+    let collection_name = path.into_inner();
+    let snapshot = form.snapshot;
+    let wait = params.wait.unwrap_or(true);
+    let timing = Instant::now();
+
+    let path = format!(
+        "{}/{}/{}",
+        dispatcher.snapshots_path(),
+        collection_name,
+        snapshot.file_name.unwrap()
+    );
+    snapshot.file.persist(&path).unwrap();
```

Comment:
We cannot panic in this part of the code, ideally those `unwrap` should be turned into validation errors.

Reply:
Noted. I'll fix that. 
Is there any resources on how to test it in a distributed cluster?